### DISCUSSION TITLE
Images not loading in NextJS 14 Headless WordPress demo fresh install

### DISCUSSION THREAD

**Discussion by LunatikDev** [vercel/next.js#62163] [Help] [answered]
### Summary

Hi everyone! I'm trying NextJS for the first time. My goal is to use it as front-end for an headless WordPress website.

I tried to deploy this template: https://vercel.com/templates/next.js/isr-blog-nextjs-wordpress

I just followed two simple steps:

Click on "Deploy"
Specified my GraphQL url from my WP install. I checked on my WP install and GraphQL seems to work there. URL is https://angrydevblog-wp.lunatik.dev/graphql
It seems images are not loaded: https://nextjs-angrydevblog.vercel.app/

I did set the env variables when i deployed on Vercel and all of the data loads fine but the images are not working :(

I try to do some searches but it seems I can't find a solution. What did I do wrong? 

NOTE: when run locally i get the following error : 
![image](https://github.com/vercel/next.js/assets/58997519/de7d8d1a-e9f7-49be-b018-616895c9ab7c)

I tried what is said there: https://nextjs.org/docs/messages/next-image-unconfigured-host but it didn't solve the problem.. here is the content of my next.config.js file:
`if (!URL.canParse(process.env.WORDPRESS_API_URL)) {
  throw new Error(`
    Please provide a valid WordPress instance URL.
    Add to your environment variables WORDPRESS_API_URL.
  `);
}

const { protocol, hostname, port, pathname } = new URL(
  process.env.WORDPRESS_API_URL,
);

/** @type {import('next').NextConfig} */
module.exports = {
  images: {
    remotePatterns: [
      {
        protocol: "https",
        hostname: "angrydevblog-wp.lunatik.dev",
        port: "",
        pathname: `${pathname}/**`,
      },
    ],
  },
};
`

**Comment by LunatikDev** [comment, marked as answer]
Ok, so i got it working, the provided next.config.js file doesn't work! 

The file provided with the project looks like this:
![image](https://github.com/vercel/next.js/assets/58997519/120d34f1-730f-4e1c-9df7-8e4058d2174a)
It should be changed to
![image](https://github.com/vercel/next.js/assets/58997519/4fff6936-4d24-4315-bcff-971b009c1ec8)
Without the parameters:

swcMinify: true, 
optimizeFonts: true, 
minimumCacheTTL: 15000000 

It will not work!

Also the hostname "secure.gravatar.com" should be added because profile avatars on latest wordpress are handled by gravatar !

**Reply by iletti** [reply]
Awesome work m8. Had the same problem. Your solution worked.

**Reply by MisterMunchkin** [reply]
Thank you! This worked for me. 

I created the project from vercel deployment using the template, and it was using the latest next version which gave me this error:

```
Image with src "http://2.gravatar.com/avatar/5f69b3eceb10d3f6a8d049ca2a63679f?s=96&d=mm&r=g" has legacy prop "layout". Did you forget to run the codemod?
Read more: https://nextjs.org/docs/messages/next-image-upgrade-to-13
```

so I had to add

      {
        protocol: 'http',
        hostname: '2.gravatar.com'
      }
on remotePatterns aswell.
